### PR TITLE
refactor: structure of lambda code

### PR DESCRIPTION
**Pull request by npalm**
- moved lambda code out of terraform modules
- use yarn workspaces with lerna to avoid duplication code

### CODE REVIEW

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -26,43 +27,44 @@ jobs:
...
-        run: echo "name=${GITHUB_REF#refs/heads/}" >> $GITHUB_OUTPUT
-      - name: Release
-        id: release
-        uses: google-github-actions/release-please-action@c078ea33917ab8cfa5300e48f4b7e6b16606aede # ratchet:google-github-actions/release-please-action@v3
-        with:
-          default-branch: ${{ steps.branch.outputs.name }}
-          release-type: terraform-module
-          token: ${{ steps.token.outputs.token }}
+
+  # release:
```
Comment (npalm) [inline .github/workflows/release.yml:38]: Disabled release part to test workflow as much as possible

File: lambdas/functions/control-plane/package.json [hunk 2]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "github-runner-lambda-scale-runners",
+  "name": "@terraform-aws-github-runner/control-plane",
```
Comment (npalm) [inline lambdas/functions/control-plane/package.json:2]: renamed module from runners to control-plan. Will rework the naming further in future PRs

**Review by PatrickMennen — APPROVED** [review]
Overall looks quite good, you can move the eslint configuration to the highest level and extend it from there in your sub-packages, just as you do with your typescript config

for the near future:
if your build and test steps will remain the same, you could integrate an nx config, which in turn will be picked up by lerna to build and test your modules

**Comment by npalm** [comment]
> Overall looks quite good, you can move the eslint configuration to the highest level and extend it from there in your sub-packages, just as you do with your typescript config
> 
> for the near future: if your build and test steps will remain the same, you could integrate an nx config, which in turn will be picked up by lerna to build and test your modules

moved eslint as well

File: .ci/Dockerfile [hunk 3]
```diff
@@ -1,26 +1,16 @@
...
+FROM build as lambdas
+COPY lambdas /lambdas
+RUN --mount=type=cache,target=/work/node_modules,id=lambdas \
+        yarn install --frozen-lockfile && yarn build
 
 FROM scratch as final
-COPY --from=runner-binaries-syncer /lambda/runner-binaries-syncer.zip /runner-binaries-syncer.zip
-COPY --from=runners                /lambda/runners.zip                /runners.zip
-COPY --from=webhook                /lambda/webhook.zip                /webhook.zip
+COPY --from=lambdas /lambdas/functions/gh-agent-syncer/runner-binaries-syncer.zip /runner-binaries-syncer.zip
```
Comment (ScottGuymer) [inline .ci/Dockerfile:14]: This is likely to trip up someone in future if a new function is added..

A better way may be to run a command in the lambdas stage to bring these zips together in a folder or even zip them..

and then copy that single file/folder here

**Review by ScottGuymer — APPROVED** [review]
Looks good. only one comment.

File: .ci/Dockerfile (hunk 3 above)
Comment (npalm) [reply .ci/Dockerfile:14]: This Dockerfile is only for convience to let users build the lambda's locally without Docker. Agree could be improved. But OK if we do this later. And create an issue?

Comment (npalm) [reply .ci/Dockerfile]: fixed

**Review by PatrickMennen — APPROVED** [review]

**Pull request merged**